src/components/AssetBreadcrumb/AssetBreadcrumb.tsx

```diff
@@ -0,0 +1,155 @@
+import { Asset } from '@cognite/sdk';
+import { Breadcrumb, Icon } from 'antd';
+import React, { useContext, useEffect, useState } from 'react';
+import styled from 'styled-components';
+import { ERROR_NO_SDK_CLIENT } from '../../constants/errorMessages';
+import { ClientSDKCacheContext } from '../../context/clientSDKCacheContext';
+import { withDefaultTheme } from '../../hoc/withDefaultTheme';
+import { AnyIfEmpty } from '../../interfaces';
+import { defaultTheme } from '../../theme/defaultTheme';
+
+export interface AssetBreadcrumbProps {
+  assetId: number;
+  maxLength?: number;
+  renderItem?: (asset: Asset, depth: number) => JSX.Element;
+  onBreadcrumbClick?: (asset: Asset, depth: number) => void;
+  theme?: AnyIfEmpty<{}>;
+}
+
+const ENTER_KEY_CODE = 13;
+
+const renderBreadcrumbs = (
+  assets: Asset[],
+  renderAsset: (asset: Asset, depth: number) => JSX.Element,
+  maxLength: number = 3
+): JSX.Element[] => {
+  const gapBreadcrumb = (
+    <Breadcrumb.Item key={'...'}>
+      <BreadcrumbSpan>...</BreadcrumbSpan>
+    </Breadcrumb.Item>
+  );
+  const assetsToRender = assets.slice();
+
+  if (assets.length > maxLength) {
+    assetsToRender.splice(maxLength - 1, assets.length - maxLength);
+  }
+
+  const renderedList = assetsToRender.map((asset, assetIndex) => (
+    <Breadcrumb.Item key={asset.id || assetIndex}>
+      {renderAsset(asset, assetIndex)}
+    </Breadcrumb.Item>
+  ));
+
+  if (assets.length > maxLength) {
+    renderedList.splice(1, 0, gapBreadcrumb);
+  }
+
+  return renderedList;
+};
+
+const AssetBreadcrumb: React.FC<AssetBreadcrumbProps> = ({
+  assetId,
+  maxLength = 3,
+  renderItem,
+  onBreadcrumbClick = () => undefined,
+}: AssetBreadcrumbProps) => {
+  const context = useContext(ClientSDKCacheContext);
+  const [assets, setAssets] = useState<Asset[]>([]);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const fetchAsset = async (id: number): Promise<Asset> => {
+    const [asset] = await context!.assets.retrieve([{ id }]);
+
+    return asset;
+  };
+
+  const renderValue = (asset: Asset, index: number): JSX.Element => {
+    if (renderItem) {
+      return renderItem(asset, index);
+    }
+
+    return (
+      <BreadcrumbSpan
+        tabIndex={0}
+        role="button"
+        onClick={() => onBreadcrumbClick(asset, index)}
+        onKeyDown={e =>
+          e.keyCode === ENTER_KEY_CODE && onBreadcrumbClick(asset, index)
+        }
+      >
+        {asset.name || asset.id}
+      </BreadcrumbSpan>
+    );
+  };
+
+  useEffect(() => {
+    let canceled = false;
+
+    if (!context) {
+      console.error(ERROR_NO_SDK_CLIENT);
+      return;
+    }
+
+    const generateAssetChain = async (id: number): Promise<void> => {
+      const assetChain: Asset[] = [];
+
+      setIsLoading(true);
+
+      let asset: Asset = await fetchAsset(id);
```

Comment:
suggestion here: can we fetch the asset.root right away as well? it'll make it slightly faster

Reply:
yes, it's two api calls that will work in parallel, that means if you have 1 -> 2 -> 3 then you can use just time * 2, not time*3

Reply:
yes, it adds some more logic, but will most likely work faster